ColorHighlighter.py

```diff
@@ -137,10 +137,40 @@ def region_name(s, is_text):
 
 def read_file(fl):
     try:
-        with codecs.open(fl, "r", "utf-8") as f:
+        with codecs.open(fl, "r", "utf_8") as f:
             return f.read()
-    except UnicodeDecodeError as err:
-        print_error("read file error in file '"+fl+"':\n" + str(err))
+    except UnicodeDecodeError:
+        # Codecs supported by Python's codec module, in order of use according
+        # to http://w3techs.com/technologies/overview/character_encoding/all
+        encodings = [
```

Comment:
Thanks. It's just that the plugin's code isn't all that great already, and I hope not to push it too far :)